### PR TITLE
Don't use streams for VoxelShape calls

### PR DESCRIPTION
**Pull request by Cryptite**
Per @electronicboy's suggestion.

Streams aren't performant with hot code and Mojang uses them quite frequently. It was suggested to try to revert stream calls in VoxelShape back to traditional for loops and Lists. Doing so seems to have increased the speed of Entity.move() from 13-22% on some cursory testing:

With 1.13.2 latest Paper (about 200.8ms per call when sampling)
image

With this patch (about 175.1ms per call when sampling)
image

The testing scenario for these profilings were just a simply small pen of about 195 or so Pigs:
image

### CODE REVIEW

File: Spigot-Server-Patches/0445-Don-t-use-streams-for-VoxelShape-calls.patch [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+@@ -2,10 +2,8 @@ package net.minecraft.server;
+ 
+ import com.google.common.collect.UnmodifiableIterator;
+ import it.unimi.dsi.fastutil.objects.Object2ByteLinkedOpenHashMap;
+-import java.util.Iterator;
+-import java.util.List;
+-import java.util.Objects;
+-import java.util.Random;
++
++import java.util.*;
```
Comment (Spottedleaf) [inline Spigot-Server-Patches/0445-Don-t-use-streams-for-VoxelShape-calls.patch]: needs a paper comment for imports, drop wildcard too

File: scripts/importmcdev.sh [hunk 2]
```diff
@@ -93,7 +93,9 @@ done
 # we do not need any lines added to this file for NMS
 
 # import FileName
-
+import VoxelShapes
```
Comment (Spottedleaf) [inline scripts/importmcdev.sh]: need to be dropped

File: Spigot-Server-Patches/0445-Don-t-use-streams-for-VoxelShape-calls.patch [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+     }
+ 
+     default boolean getCubes(@Nullable Entity entity, AxisAlignedBB axisalignedbb) {
+diff --git a/src/main/java/net/minecraft/server/StreamAccumulator.java b/src/main/java/net/minecraft/server/StreamAccumulator.java
+deleted file mode 100644
+index 1f70c0405..000000000
+--- a/src/main/java/net/minecraft/server/StreamAccumulator.java
++++ /dev/null
+@@ -1,44 +0,0 @@
+-package net.minecraft.server;
```
Comment (Spottedleaf) [inline Spigot-Server-Patches/0445-Don-t-use-streams-for-VoxelShape-calls.patch]: keep this around in case plugins use it

**Comment by Spottedleaf** [comment]
try to keep the ABI intact (i.e convert stream params to list for param forwarding)
for return type changes do the same but have the new method keep a different name

and //paper comments need to be added

**Comment by ninja-** [comment]
the last thing mojang cares about is performance. nice find 👍

**Comment by Foorcee** [comment]
Is it perhaps also a problem in the 1.14?

**Comment by ninja-** [comment]
> Is it perhaps also a problem in the 1.14?

bet

**Comment by zachbr** [comment]
Still lots of broken ABI in this PR

**Comment by Cryptite** [comment]
I'm unlikely to revisit this patch for 1.13.2. Welcome to close it or let somebody else take over but I'd rather redo the work in 1.14 or such rather than be behind.

**Comment by Spottedleaf** [comment]
I've found that this patch is not as relevant in 1.14

**Comment by Hyronymos-Old** [comment]
https://sparkprofiler.github.io/#plFv0FVL4R

It looks like for me that voxelshape is cusing laggs in 1.14.4, isnt it?

**Pull request closed without merging**